### PR TITLE
Data example

### PR DESCRIPTION
**Pull request by clarkfitzg**
This shows an example of some simple analysis where the data set is large enough for distributed data structures to be useful.

### CODE REVIEW

**Comment by lawremi** [comment]
This is a great start. Since it's a complex discussion, I'd like to chat about this. I'm available Tuesday at 3 PM. Everyone else is of course welcome to join. Btw, did you feel the earthquake?

**Comment by clarkfitzg** [comment]
That time works for me. I just sent an invite.

Yes, it was quite an earthquake!

File: examples/pems_data/pems_ddR.R [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+library(ddR)
+
+# Trying various options with ddR: useBackend(parallel, type = 'PSOCK')
+
+# There are 4 gzipped files here
+station_files = list.files("~/data/pems/", full.names = TRUE)
+
+read1 <- function(file) {
+    read.table(station_files[1], header = FALSE, sep = ",",
```
Comment (clarkfitzg) [inline examples/pems_data/pems_ddR.R]: Should be `file`, not `station_files[1]`.

**Comment by clarkfitzg** [comment]
## Timing

```
$ ./timer.sh
Sequential version base R
real    3m32.358s
user    3m25.768s
sys     0m6.585s

R's standard parallel library
real    1m19.140s
user    4m1.387s
sys     0m12.280s

ddR
real    148m48.075s
user    83m20.409s
sys     227m43.427s
```

So it looks like we get the typical 3x speedup with `parallel` compared to single threaded R, while ddR is two orders of magnitude slower.
This is using the current CRAN version of ddR. My recent PR is mostly reorganization, so there's no reason to think it would improve things. 

## Memory

This is running on a 2016 Macbook pro with 16 GB memory.

| Version | Max Memory Usage | After `gc()` |
| --- | --- | --- |
| Sequential | 5.5 GB | 3.50 GB |
| Parallel | 5.5 GB | 3.47 GB |
| ddR | 15 - 20 GB (at least) |  |

Running the ddR version required swapping, and much of the slowness can probably be attributed to that.

File: examples/pems_data/pems_parallel.R [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+# Parallelized version of cut
+pcut <- function(x, breaks, labels = NULL, mc.cores = ncores) {
+    localcut <- function(y) {
+        table(cut(y, breaks = breaks, labels = labels))
+    }
+    n <- length(x)
+    # For `fork` systems it makes sense to have data structures like the
+    # one created here: the base objects augmented with a split for
+    # parallelization
```
Comment (clarkfitzg) [inline examples/pems_data/pems_parallel.R]: A suggestion on an alternative implementation for `fork` backend.

**Comment by clarkfitzg** [comment]
Ran this script again this time limiting it to 4 million rows total to alleviate the memory pressure. This helped ddR quite a lot, but it's still 2-3x slower than regular sequential R. A performance goal would be between 

## Timing

```
$ ./timer.sh
Sequential version base R
real    0m20.833s
user    0m20.352s
sys     0m0.437s

R's standard parallel library
real    0m7.289s
user    0m21.590s
sys     0m0.941s

ddR
real    0m51.402s
user    2m45.732s
sys     0m16.824s
```

## Memory

This is running on a 2016 Macbook pro with 16 GB memory and 4 physical cores.

| Version | Max Memory Usage | After `gc()` |
| --- | --- | --- |
| Parallel | 505 MB | 393 MB |
| ddR | 1420 MB  | 1420 MB |

**Pull request merged**